demos/python_demos/colorization/colorization.py

```diff
@@ -0,0 +1,138 @@
+
+
+def get_model_files(args_in):
+    config_xml_in = args_in.model
+    model_path = os.path.splitext(config_xml_in)[0]
+    weights_bin_in = model_path + ".bin"
+    level_up = os.path.split(os.path.split(model_path)[0])[0]
+    coeffs_in = os.path.join(level_up, os.path.split(level_up)[1]) + ".npy"
+    return config_xml_in, weights_bin_in, coeffs_in
+
+
+if __name__ == '__main__':
+    args = build_arg().parse_args()
+    config_xml, weights_bin, coeffs = get_model_files(args)
+
+    # scale and mean are stored in the source caffemodel
+    scale, mean = get_parameters(config_xml)
+
+    log.basicConfig(format="[ %(levelname)s ] %(message)s",
+                    level=log.INFO if not args.verbose else log.DEBUG, stream=sys.stdout)
+
+    log.debug("Load network")
+    load_net = IENetwork(model=config_xml, weights=weights_bin)
+    exec_net = IECore().load_network(network=load_net, device_name=args.device)
+
+    assert len(load_net.inputs) == 1, "Number of inputs does not match network inputs"
+    input_blob = next(iter(load_net.inputs))
+    input_shape = load_net.inputs[input_blob].shape
+    assert input_shape == [1, 1, 224, 224], "Shape of inputs does not match network shape inputs"
+
+    assert len(load_net.outputs) == 1, "Number of outputs does not match network outputs"
+    output_blob = next(iter(load_net.outputs))
+    output_shape = load_net.outputs[output_blob].shape
+    assert output_shape == [1, 313, 56, 56], "Shape of outputs does not match network shape outputs"
+
+    n, c, h_in, w_in = input_shape
+
+    cap = cv.VideoCapture(args.input if args.input else 0)
+    while cv.waitKey(1) < 0:
+        log.debug("#############################")
+        hasFrame, original_frame = cap.read()
+        if not hasFrame:
+            break
+        (h_orig, w_orig) = original_frame.shape[:2]
+
+        log.debug("Preprocessing frame")
+        if original_frame.shape[2] > 1:
+            frame = cv.cvtColor(cv.cvtColor(original_frame, cv.COLOR_BGR2GRAY), cv.COLOR_GRAY2RGB)
+        else:
+            frame = cv.cvtColor(original_frame, cv.COLOR_GRAY2RGB)
+
+        img_rgb = frame.astype(np.float32) / scale
+        img_lab = cv.cvtColor(img_rgb, cv.COLOR_RGB2Lab)
+        img_l = img_lab[:, :, 0]
+
+        img_rs = cv.resize(img_rgb, (w_in, h_in))
+        img_lab_rs = cv.cvtColor(img_rs, cv.COLOR_RGB2Lab)
+        img_l_rs = img_lab_rs[:, :, 0]
+        img_l_rs -= mean
+
+        log.debug("Network inference")
+        res = exec_net.infer(inputs={input_blob: [img_l_rs]})
+
+        (n_out, c_out, h_out, w_out) = res[output_blob].shape
+        update_res = np.zeros((n_out, 2, h_out, w_out)).astype(np.float32)
+        color_coeff = np.load(coeffs)
+
+        for res_blob, color_coeff_blob in list(zip(res[output_blob][0, :, :, :], color_coeff)):
+            for upd_res_blob, clr_coeff in list(zip(update_res[0, :, :, :], color_coeff_blob)):
+                upd_res_blob += res_blob * clr_coeff
```

Comment:
You can replace these loops with a single line: `update_res = (res[output_blob] * color_coeff.transpose()[:, :, np.newaxis, np.newaxis]).sum(1)`.

Reply:
@Wovchena done)

Reply:
`update_res` is always used as `update_res[0, ...]`. I suggest omitting this dimension. There is a connected note in https://github.com/opencv/open_model_zoo/pull/356#discussion_r356046159 about batches.

Reply:
@Wovchena done)